### DISCUSSION TITLE
Signing commits with GitHub and Renovate Bot

### DISCUSSION THREAD

**Discussion by eana** [renovatebot/renovate#12692] [Request Help] [answered]
Hi there!

I configured renovate to sign the git commits, but unfortunately, I can't get it to work -- It is worth mentioning that I have a similar approach working in GitLab.

To provide a better context, I will try to be as exact as possible with all the steps I carried out.

1. I generated a GPG key following this document: https://docs.gitlab.com/ee/user/project/repository/gpg_signed_commits/#generating-a-gpg-key
2. I exported the public key using the email entered in the step above `gpg --armor --export <Email>`
3. I added the GPG key to the github account the renovate-bot will use
4. I configure renovate to use the GPG private key and run it

```bash
$ git clone https://github.com/bolawell/renovate-bot.git && cd renovate-bot

$ gpg --output private.pgp --armor --export-secret-key <Email>

$ docker run -it --name node --rm -v $(pwd):/data node:17 bash

# cd /data

# gpg --import private.pgp

# rm -f private.pgp

# git config --global user.email "<Email>"
# git config --global user.name "Renovate Bot"
# git config --global commit.gpgsign true
# git config --global user.signingkey <Email>

# export RENOVATE_TOKEN="<Personal access token>"
# export GITHUB_COM_TOKEN="<Personal access token>"
# export LOG_LEVEL=debug

# npm install
# npm run renovate
```

5. Test if the sign commits are really working I committed and pushed a dummy change from the container I am running renovate

```bash
# cd /
# git clone https://github.com/bolawell/test-repo.git                
# cd /test-repo
# echo "## Test signed commits" >> Dockerfile
# git add .
# git commit -v -m "Test signed commits"
# git push
```
6. Interestingly, when verifying the commit, the commit is signed:

```bash
# git verify-commit HEAD
gpg: Signature made Tue Nov 16 12:10:33 2021 UTC
gpg:                using RSA key C3FC223B6CEC60E4857141EABC59BFF014346A0C
gpg:                issuer "<Email>"
gpg: Good signature from "Renovate Bot <Email>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: C3FC 223B 6CEC 60E4 8571  41EA BC59 BFF0 1434 6A0C
```

Here is the [config.js](https://github.com/bolawell/renovate-bot/blob/master/config.js) file.

What am I doing wrong?

Thank you

**Comment by viceice** [comment, marked as answer]
You need to configure  like this: https://docs.renovatebot.com/self-hosted-configuration/#gitprivatekey

**Reply by eana** [reply]
Thank you, @viceice! I got it working. In case anyone else has this issue, this is what I did.

1. I added `gitPrivateKey: process.env.GIT_PRIVATE_KEY` to my `config.js`
2. `export GIT_PRIVATE_KEY=$(cat private.pgp)`
3. Run renovate as user.

`gpg --import private.pgp` was not necessary anymore.
